cmd/kubeadm/app/constants/constants.go

```diff
@@ -181,6 +181,8 @@ const (
 	PatchNodeTimeout = 2 * time.Minute
 	// TLSBootstrapTimeout specifies how long kubeadm should wait for the kubelet to perform the TLS Bootstrap
 	TLSBootstrapTimeout = 2 * time.Minute
+	// PullImageRetry specifies how many times ContainerRuntime retries when pulling image failed
+	PullImageRetry = 5
```

Comment:
when we originally discussed this there were also comments about using exponential backoff.
i'm fine with using 5 retries, but let's see if anyone has more comments.